### PR TITLE
Implement React-Express API

### PR DESCRIPTION
**Pull request by loganb294**
This pull request focuses on setting up a React-Express API to interact with a render database. The primary goal is to establish the foundational components and ensure that initial development is in place for future work.

working files:
most / all 

Relevant User Stories:
[HERE](https://github.com/BIT-Studio-3/group-project-24-2-s-m-l-l/issues/87)

Relevant Issues:

[epic issue](https://github.com/BIT-Studio-3/group-project-24-2-s-m-l-l/issues/90)
[API issue](https://github.com/BIT-Studio-3/group-project-24-2-s-m-l-l/issues/98)

### CODE REVIEW

**Comment by loganb294** [comment]
@MikeBike2 you are primary reviewer

**Comment by LoganHIT** [comment]
Could you guys specify the files we need to review, or are all of these needing review?

**Comment by loganb294** [comment]
> Could you guys specify the files we need to review, or are all of these needing review?

this is the exact list of files needing reviewed, 
@LoganHIT @MikeBike2 

app.js
controllers/v1/authController.js
controllers/v1/base.js
controllers/v1/user.js
middleware/authMiddleware.js
middleware/validation.js
prisma/prisma.js
prisma/schema.prisma
routes/v1/base.js
routes/v1/user.js
utils/statusCode.js

**Comment by LoganHIT** [comment]
Thanks fella, helps trim down the reading a bit

**Comment by MikeBike2** [comment]
i will get my review in tommorrow

File: app.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import cors from "cors"
+
+
+
+// Create an Express application
+const app = express();
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+app.use(cors())
+
```
Comment (MikeBike2) [inline app.js]: what does the cors part do ? don't remember coming across this

**Review by MikeBike2 — CHANGES_REQUESTED** [review]
looks fine to me, could you just explain what the cors import does ? don't remember seeing this in graysons labs

File: app.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const PORT = process.env.PORT || 3000;
+
+// Swagger options
+const swaggerOptions = {
+  definition: {
+    openapi: "3.0.0",
+    info: {
+      title: "Student Management System API",
+      version: "1.0.0",
+      description: "A student management system API",
```
Comment (LoganHIT) [inline app.js]: 'student management system API" ? are we sure

File: app.js (hunk 1 above)
Comment (loganb294) [reply app.js]: basically it meant to make it so people can only add users from the login/register page

**Review by LoganHIT — APPROVED** [review]
Overall I think, with my limited knowledge it looks fine. 
Maybe in future add more comments explaining sections, describing where things relate and what they do.

Ready for merging I believe.

**Review by MikeBike2 — APPROVED** [review]
looks fine mostly boiler plate code here from graysons lab so have no issues here.

**Pull request merged**